tools/makecsr/make-ev-csr.go

```diff
@@ -23,16 +25,20 @@ import (
 func main() {
 	var (
 		keyLabel string
-		// ou       string
-		cn string
-		// email    string
 	)
 	flag.StringVar(&keyLabel, "l", "mykey", "Label of the key in the HSM")
-	// flag.StringVar(&ou, "ou", "Mozilla AMO Production Signing Service", "OrganizationalUnit of the Subject")
-	// flag.StringVar(&cn, "cn", "Content Signing Intermediate", "CommonName of the Subject")
-	// flag.StringVar(&email, "email", "foxsec@mozilla.com", "Email of the Subject")
 	flag.Parse()
 
+	// We need the email address encoded as asn1.ia5
+	email_utf8 := "release+certificates@mozilla.com"
+	type ia5String struct {
+		A string `asn1:"ia5"`
+	}
+	email_ia5, err := asn1.Marshal(ia5String{email_utf8})
```

Comment:
I don't think you should marshal here. Instead, try setting the Value on line 76 to `ia5String{email_utf8}`

Reply:
My understanding is that marshal does the actual encoding. Until then, it's a normal golang string with a tag.

I used [this tool](https://lapo.it/asn1js/#) to verify that the field is identified as ia5 encoding in the output "csr". (The output is not a valid csr, but it is valid asn1.)

I'll test my understanding later today.

Reply:
The call to `xz509.CreateCertificateRequest` later will take care of `asn1.Marshal` for the entire subject.

Reply:
Nope. With the direct assignment, it decodes as:
```
SEQUENCE (2 elem)
          OBJECT IDENTIFIER 1.2.840.113549.1.9.1 emailAddress (PKCS #9. Deprecated, use an altName extension instead)
          UTF8String release+certificates@mozilla.com
```

With the call to marshal, it decodes as:
```
SEQUENCE (2 elem)
          OBJECT IDENTIFIER 1.2.840.113549.1.9.1 emailAddress (PKCS #9. Deprecated, use an altName extension instead)
          OCTET STRING (1 elem)
            SEQUENCE (1 elem)
              IA5String release+certificates@mozilla.com
```

But now I see I wasn't using the full output of the tool -- I'll go look for `altName` OID now